packages/roosterjs-editor-core/lib/corePlugins/utils/InlineEntityHandlers/inlineEntityOnPluginEvent.ts

```diff
@@ -24,11 +38,68 @@ export function inlineEntityOnPluginEvent(event: PluginEvent) {
 
                     // If the entity removed is a readonly entity, try to remove delimiters around it.
                     if (isReadOnly(entity)) {
-                        removeDelimiters(wrapper);
+                        removeDelimiters(entity.wrapper);
                     }
                     break;
             }
             break;
+
+        case PluginEventType.KeyDown:
+        case PluginEventType.KeyUp:
```

Comment:
I think just handling KeyDown event should be enough

Reply:
Removed KeyUp as suggested